### PR TITLE
Give wiggle room for selecting SLA policy names if timestamps are off by milliseconds 

### PR DESCRIPTION
**Pull request by fivetran-jamie**
## PR Overview
**This PR will address the following Issue/Feature:**
Customer has reported seeing records in `zendesk__sla_policies` with null `sla_policy_name` values when they should not be null. We've isolated this problem to this [join](https://github.com/fivetran/dbt_zendesk/blob/main/models/sla_policy/int_zendesk__sla_policy_applied.sql#L49-L50). It turns out that some records can come from Zendesk in the wrong order by a margin of a couple of milliseconds, which our current join logic is too strict to include. 

**This PR will result in the following new package version:**

v0.18.1

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

## Bug Fix
- Addressed an issue in which some records in `zendesk__sla_policies` might erroneously have a null `sla_policy_name` due to system-generated millisecond-long gaps in timestamps. The package now compares timestamps to the nearest `second` when selecting valid SLA policy names in [int_zendesk__sla_policy_applied](https://github.com/fivetran/dbt_zendesk/blob/main/models/sla_policy/int_zendesk__sla_policy_applied.sql).

## Under the Hood
- Updated `consistency_sla_policies` data validation test to account for the above change.
- Updated logic in `sla_count_match` data validation test.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [x] dbt run (if incremental models are present) && dbt test

Before marking this PR as "ready for review" the following have been applied:
- [n/a] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
- [x] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [x] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

Added `fivetran_integrity_sla_count_match_tickets` exclusion tickets related to https://github.com/fivetran/dbt_zendesk/issues/175 and validation tests are passing using both the testing and prod schemas:
image

Regarding the `add_exceptions` CTE I [added](https://github.com/fivetran/dbt_zendesk/pull/174/files#diff-0d0d9ef44ed419fb414f050ad80372140d4d5a8a51d3d35b65bb301e10c9b10d) to the `consistency_sla_policies` test - I added the `name_was_null_prior` clause because there were ~5K tickets in our internal dataset that had their null SLA policy names updated by my fix.

### If you had to summarize this PR in an emoji, which would it be?
 
🪟

### CODE REVIEW

File: integration_tests/tests/consistency/consistency_sla_policies.sql
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply integration_tests/tests/consistency/consistency_sla_policies.sql:1]: Some validation tests (like the sla_metrics_parity and sla_first_reply_time_match tests) don't make much sense when we aren't using business hours so we can probably disable that test for now when the schedules variable is disabled.

**Review by fivetran-joemarkiewicz — APPROVED** [review]
@fivetran-jamie thanks for working so quick to address this issue. I tested these changes and they look good to me. I do have a suggestion to make some adjustments to the validation tests to ensure the either don't run or ignore the business metrics when the `using_schedules` variable is disabled.

If you could make those updates so we could use these validations in the future that would be great. Once those are applied, you can move this forward to release review. Thanks again!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
+# dbt_zendesk v0.18.1
+[PR #174](https://github.com/fivetran/dbt_zendesk/pull/174) includes the following changes:
+
+## Bug Fix
+- Addressed an issue in which some records in `zendesk__sla_policies` might erroneously have a null `sla_policy_name` due to system-generated millisecond-long gaps in timestamps. The package now compares timestamps to the nearest `second` when selecting valid SLA policy names in [int_zendesk__sla_policy_applied](https://github.com/fivetran/dbt_zendesk/blob/main/models/sla_policy/int_zendesk__sla_policy_applied.sql).
+
+## Under the Hood
+- Updated `consistency_sla_policies` data validation test to account for the above change.
```
Comment (fivetran-avinash) [inline CHANGELOG.md:8]: We can probably combine the two sentences here into one, I think they were both to account for the above changes?

File: integration_tests/tests/consistency/consistency_sla_policies.sql [hunk 2]
```diff
@@ -60,8 +60,29 @@ final as (
...
     from dev_not_in_prod
+),
+
+final as (
+    select 
+        *,
+        max(sla_elapsed_time) over (partition by ticket_id, metric, sla_applied_at) as max_sla_elapsed_time,
+        min(sla_elapsed_time) over (partition by ticket_id, metric, sla_applied_at) as min_sla_elapsed_time,
+
+        {#
```
Comment (fivetran-avinash) [inline integration_tests/tests/consistency/consistency_sla_policies.sql:71]: Reminder to remove before merging!

File: integration_tests/tests/consistency/consistency_sla_policies.sql [hunk 3]
```diff
@@ -60,8 +60,29 @@ final as (
...
 
 select *
 from final
-{{ "where ticket_id not in " ~ var('fivetran_consistency_sla_policies_exclusion_tickets',[]) ~ "" if var('fivetran_consistency_sla_policies_exclusion_tickets',[]) }}
\ No newline at end of file
+where 
+    {# Take differences in runtime into account #}
+    max_sla_elapsed_time - min_sla_elapsed_time > 2 
+    
+    {# Remove after v0.18.1 #}
```
Comment (fivetran-avinash) [inline integration_tests/tests/consistency/consistency_sla_policies.sql:87]: Reminder to remove before merging!

File: integration_tests/tests/consistency/consistency_sla_policies.sql
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply integration_tests/tests/consistency/consistency_sla_policies.sql:1]: @fivetran-jamie were these suggestions applied? Didn't see these conditional blocks added.

**Review by fivetran-avinash — COMMENTED** [review]
@fivetran-jamie One question ([see here](https://github.com/fivetran/dbt_zendesk/pull/174#discussion_r1815266233)) and a few minor comments before approving.

File: integration_tests/tests/consistency/consistency_sla_policies.sql (hunk 2 above)
Comment (fivetran-jamie) [reply integration_tests/tests/consistency/consistency_sla_policies.sql]: removed

File: CHANGELOG.md (hunk 1 above)
Comment (fivetran-jamie) [reply CHANGELOG.md]: combined

File: integration_tests/tests/consistency/consistency_sla_policies.sql (hunk 3 above)
Comment (fivetran-jamie) [reply integration_tests/tests/consistency/consistency_sla_policies.sql]: removed

File: integration_tests/tests/consistency/consistency_sla_policies.sql
[the remark this answers is not in the document]
Comment (fivetran-jamie) [reply integration_tests/tests/consistency/consistency_sla_policies.sql:1]: Talked offline at standup and agreed that these updates weren't necessary -- we do actually have the `schedule` table in our dataset and therefore don't need to run validation tests with `using_schedules` disabled

**Review by fivetran-avinash — APPROVED** [review]
lgtm

**Pull request merged**